### ISSUE TITLE
Coffeescript big: single quoted string with tagged template literal

### ISSUE THREAD

**Issue by greghuc** [DanielXMoore/Civet#311]
### Coffeescript bug (parse error)

```
"civet coffeeCompat"

upperCaseExpr = (textParts, expressions...) ->
  textParts.reduce (text, textPart, i) ->
    text + expressions[i - 1].toUpperCase() + textPart

greet = (_name, _adjective) ->
  upperCaseExpr'A string with no variables'

console.log greet("greg", "awesome")
```
gives
```
"civet coffeeCompat"

upperCaseExpr = (textParts, expressions...) ->
  textParts.reduce (text, textPart, i) ->
    text + expressions[i - 1].toUpperCase() + textPart

greet = (_name, _adjective) ->
  upperCaseExpr'A string with no variables'
               ^ ParseError
```

### Playground links

[Coffeescript link](https://coffeescript.org/#try:%22civet%20coffeeCompat%22%0A%0AupperCaseExpr%20%3D%20(textParts%2C%20expressions...)%20-%3E%0A%20%20textParts.reduce%20(text%2C%20textPart%2C%20i)%20-%3E%0A%20%20%20%20text%20%2B%20expressions%5Bi%20-%201%5D.toUpperCase()%20%2B%20textPart%0A%0Agreet%20%3D%20(_name%2C%20_adjective)%20-%3E%0A%20%20upperCaseExpr'A%20string%20with%20no%20variables'%0A%0Aconsole.log%20greet(%22greg%22%2C%20%22awesome%22)%0A)
[Civet link](https://civet.dev/playground/?code=ImNpdmV0IGNvZmZlZUNvbXBhdCIKCnVwcGVyQ2FzZUV4cHIgPSAodGV4dFBhcnRzLCBleHByZXNzaW9ucy4uLikgLT4KICB0ZXh0UGFydHMucmVkdWNlICh0ZXh0LCB0ZXh0UGFydCwgaSkgLT4KICAgIHRleHQgKyBleHByZXNzaW9uc1tpIC0gMV0udG9VcHBlckNhc2UoKSArIHRleHRQYXJ0CgpncmVldCA9IChfbmFtZSwgX2FkamVjdGl2ZSkgLT4KICB1cHBlckNhc2VFeHByJ0Egc3RyaW5nIHdpdGggbm8gdmFyaWFibGVzJwoKY29uc29sZS5sb2cgZ3JlZXQoImdyZWciLCAiYXdlc29tZSIp)

**Comment by greghuc** [comment]
Use of [tripped quoted string works fine](https://civet.dev/playground/?code=ImNpdmV0IGNvZmZlZUNvbXBhdCIKCnVwcGVyQ2FzZUV4cHIgPSAodGV4dFBhcnRzLCBleHByZXNzaW9ucy4uLikgLT4KICB0ZXh0UGFydHMucmVkdWNlICh0ZXh0LCB0ZXh0UGFydCwgaSkgLT4KICAgIHRleHQgKyBleHByZXNzaW9uc1tpIC0gMV0udG9VcHBlckNhc2UoKSArIHRleHRQYXJ0CgpncmVldCA9IChfbmFtZSwgX2FkamVjdGl2ZSkgLT4KICB1cHBlckNhc2VFeHByIiIiQSBzdHJpbmcgd2l0aCBubyB2YXJpYWJsZXMiIiIKCmNvbnNvbGUubG9nIGdyZWV0KCJncmVnIiwgImF3ZXNvbWUiKQ%3D%3D):

```
"civet coffeeCompat"

upperCaseExpr = (textParts, expressions...) ->
  textParts.reduce (text, textPart, i) ->
    text + expressions[i - 1].toUpperCase() + textPart

greet = (_name, _adjective) ->
  upperCaseExpr"""A string with no variables"""

console.log greet("greg", "awesome")
```

I believe the underlying issue is that Coffeescript is converting both the single-quoted string and triple-quoted string to a template literal. Civet seems to convert single-quoted to a double-quoted string, and triple-quoted to a template literal.

And a quick check with the Chrome dev tools console shows that the Javascript tagged template literal function only wants a template literal, not a single or double quoted string (grabbing [example from MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#raw_strings)):

```
function tag(strings) {
  console.log(strings.raw[0]);
}

tag`string text line 1 \n string text line 2`;
=> string text line 1 \n string text line 2

tag"string text line 1 \n string text line 2";
=> Uncaught SyntaxError: Unexpected string
tag'string text line 1 \n string text line 2';
=> Uncaught SyntaxError: Unexpected string
```

**Comment by edemaine** [comment]
Indeed, `'strings'` are currently not able to become template literals, while `'''strings'''` and `"""strings"""` always are and `"strings"` are in coffeeCompat mode. 

We should be able to fix this, but it will require some more tweaking to separation between string literals and template literals previously considered in #294.  

Meanwhile, please use `"strings"` or `'''strings'''`.

**Comment by STRd6** [comment]
@edemaine Do you see any downside to allowing all types of strings to be template literals as a default?

**Comment by edemaine** [comment]
No, I don't think so, which is why CoffeeScript made this decision. So I was thinking of just adding `StringLiteral` to the tagged template literal rule, and converting the string to back ticks. (Actually we may need to do this already with CoffeeScript double-quote strings that have no expansions.) Feel free to do this, or I can.

**Comment by STRd6** [comment]
You can. I'm looking into `||>` right now.
